From: Jean Delvare <jdelvare@suse.de>
Subject: perf/x86/intel: Handle Broadwell family processors
Patch-mainline: Never, different implementation
References: bsc#1093183

While we don't want to support the Broadwell family processors fully
on such an old kernel, we should at least not crash on them. So
enable the late ack feature for all Broadwell family processors, as
we did for Haswell family processors already.

Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 arch/x86/kernel/cpu/perf_event_intel.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

--- a/arch/x86/kernel/cpu/perf_event_intel.c
+++ b/arch/x86/kernel/cpu/perf_event_intel.c
@@ -2149,7 +2149,7 @@ __init int intel_pmu_init(void)
 			x86_pmu.event_constraints = intel_gen_event_constraints;
 			pr_cont("generic architected perfmon, ");
 
-			/* HSW late ack, bsc#929142 */
+			/* HSW/BDW late ack, bsc#929142 and bsc#1093183 */
 			switch (boot_cpu_data.x86_model) {
 	        	case 60: /* 22nm Haswell Core */
         		case 63: /* 22nm Haswell Server */
@@ -2157,6 +2157,14 @@ __init int intel_pmu_init(void)
         		case 70: /* 22nm Haswell + GT3e (Intel Iris Pro graphics) */
 				pr_cont("Haswell pmu late ack, ");
 				x86_pmu.late_ack = true;
+				break;
+			case 61: /* 14nm Broadwell Core */
+			case 71: /* 14nm Broadwell + GT3e (Intel Iris Pro graphics) */
+			case 79: /* 14nm Broadwell Server */
+			case 86: /* 14nm Broadwell Xeon D */
+				pr_cont("Broadwell pmu late ack, ");
+				x86_pmu.late_ack = true;
+				break;
 			}
 			break;
 		}
